php/libraries/User.class.inc

```diff
@@ -532,5 +532,27 @@ class User extends UserPermissions
             (string) $this->userInfo['Password_hash']
         );
     }
+
+
+    /**
+     * Return the last login of this user to the given database.
+     *
+     * @param \Database $DB The database connection to check
+     *
+     * @return string The date in YYYY-MM-DD format.
+     */
+    function getLastLogin(\Database $DB)
```

Comment:
I think this is going to generate a phan error. pselectOne has a `mixed` return type (array, string, or false).

It would be good to add a proper return type here. Bonus points if it's a DateTime object

Reply:
I think changing it to a DateTime object would break the code which is currently using it, but I agree that that should be a future change

Reply:
it was pretty easy to change here and just reformat in the one place that uses it, I'll deal with the fallout of that on #5896 when I eventually rebase it onto master.